From fd4e8693980e47385a543e7b002eec0b88bd09df Mon Sep 17 00:00:00 2001
Message-ID: <fd4e8693980e47385a543e7b002eec0b88bd09df.1690543454.git.lorenzo@kernel.org>
From: Lorenzo Bianconi <lorenzo@kernel.org>
Date: Tue, 18 Jul 2023 14:05:50 +0200
Subject: [PATCH net-next 1/2] net: ethernet: mtk_eth_soc: enable page_pool
 support for MT7988 SoC

In order to recycle pages, enable page_pool allocator for MT7988 SoC.

Tested-by: Daniel Golle <daniel@makrotopia.org>
Signed-off-by: Lorenzo Bianconi <lorenzo@kernel.org>
---
 drivers/net/ethernet/mediatek/mtk_eth_soc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/mediatek/mtk_eth_soc.c
+++ b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
@@ -1755,7 +1755,7 @@ static void mtk_update_rx_cpu_idx(struct
 
 static bool mtk_page_pool_enabled(struct mtk_eth *eth)
 {
-	return eth->soc->version == 2;
+	return mtk_is_netsys_v2_or_greater(eth);
 }
 
 static struct page_pool *mtk_create_page_pool(struct mtk_eth *eth,
